src/main/kotlin/org/rust/ide/annotator/RsHighlightingAnnotator.kt

```diff
@@ -51,17 +51,40 @@ class RsHighlightingAnnotator : AnnotatorBase() {
         }
 
         return when (element.elementType) {
+            DOLLAR -> RsColor.MACRO // TODO: It would be possible to see whether parent is RsMacroBindingGroups or RsMacroExpansionReferenceGroup or RsMacroBinding
             IDENTIFIER, QUOTE_IDENTIFIER, SELF -> highlightIdentifier(element, parent, holder)
             // Although we remap tokens from identifier to keyword, this happens in the
             // parser's pass, so we can't use HighlightingLexer to color these
             in RS_CONTEXTUAL_KEYWORDS -> RsColor.KEYWORD
             FLOAT_LITERAL -> RsColor.NUMBER
 
-            Q -> if (parent is RsTryExpr) {
+            Q -> if (parent is RsMacroExpansionReferenceGroup || parent is RsMacroBindingGroup) { // TODO: and it the last non-whitespace element in parent
+                RsColor.MACRO
+            } else if (parent is RsTryExpr) {
                 RsColor.Q_OPERATOR
             } else {
                 null
             }
+            MUL -> if (parent is RsMacroExpansionReferenceGroup || parent is RsMacroBindingGroup) { // TODO: and it is the last non-whitespace element in parent
+                RsColor.MACRO
+            } else {
+                null
+            }
+            PLUS -> if (parent is RsMacroExpansionReferenceGroup || parent is RsMacroBindingGroup) { // TODO: and it is the last non-whitespace element in parent
+                RsColor.MACRO
+            } else {
+                null
+            }
+            COLON -> if (parent is RsMacroBinding) {
+                RsColor.MACRO
+            } else {
+                null
+            }
+            LPAREN, LBRACE, RPAREN, RBRACE -> if (parent is RsMacroBindingGroup || parent is RsMacroExpansionReferenceGroup || parent is RsMacroBindingGroup) {
```

Comment:
For follow-up changes, should I amend the commit and force push, or just add new commits on top?

Reply:
We use both approaches, actually, and each of them has cons:
- it's harder to review new changes in case of force pushes (especially if PR is large)
- "new commits" approach usually requires come history clean up (i.e. squashing and force pushing at the end of review) to avoid some "fixup" commits

Since this PR is quite small, it's up to you